osu.Server.QueueProcessor/QueueProcessor.cs

```diff
@@ -100,9 +99,10 @@ public void Run(CancellationToken cancellation = default)
                                 continue;
                             }
 
-                            var redisValue = database.ListRightPop(inputQueueName);
+                            var redisItems = database.ListRightPop(inputQueueName, config.BatchSize);
 
-                            if (!redisValue.HasValue)
+                            // null or empty check is required for redis 6.x. 7.x reports `null` instead.
+                            if (redisItems == null || redisItems.Length == 0 || redisItems[0].IsNullOrEmpty)
```

Comment:
Is it possible for any other member to _not_ be null or empty? So if element 0 was null or empty but element 1 wasn't.

That is, could this be skipping items? Maybe a better way of handling is something like `redisItems = redisItems.Where(i => !i.IsNullOrEmpty).ToArray()`.

Reply:
I don't... think so. But best to be safe so I'll change it.